core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -557,17 +555,31 @@ object CachedDataFrame
     m
   }
 
+  // Some helper fields to show correct plan Info on UI as
+  // per current set of constants even though a cached plan
+  // is executed
+  var pls: Map[Int, ParamLiteral] = null
```

Comment:
As noted in general email, never use scala immutable maps for frequent lookups/writes. Change to IntObjectHashMap.withExpectedSize(8) or something and change to val. Also the visibility of this and the one below can be private. A better name like "paramLiterals" will be better.

Reply:
Ok.